### PR TITLE
[RPC][wallet] Add `rescanblockchain` RPC command functionality

### PR DESCRIPTION
**Pull request by furszy**
Added the following changes:

1) The wallet scan chain for transactions process will now return null if scan was successful. Otherwise, if a complete rescan was not possible (due to corruption or abort), returns pointer to the most recent block that could not be scanned.
2) New `rescanblockchain` RPC command with two arguments `start_height` `end_height` to trigger a chain rescan at any time during runtime + test case (adaptation of #7061).
3) New `abortrescan` RPC command to stop any long running rescan if needed at runtime (this is helpful for example in case of importing a private/sapling key that if the user is not aware of the extra "scan" boolean argument, it will rescan the complete chain). Coming from #10208.

### CODE REVIEW

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -4179,6 +4179,67 @@ UniValue getsaplingnotescount(const JSONRPCRequest& request)
     return count;
 }
 
+UniValue rescanblockchain(const JSONRPCRequest& request)
+{
+    if (request.fHelp || request.params.size() > 2) {
+        throw std::runtime_error(
+                "rescanblockchain (\"start_height\") (\"stop_height\")\n"
```
Comment (Fuzzbawls) [inline src/wallet/rpcwallet.cpp:4186]: no quotes around numeric arguments

File: src/wallet/rpcwallet.cpp [hunk 2]
```diff
@@ -4179,6 +4179,67 @@ UniValue getsaplingnotescount(const JSONRPCRequest& request)
...
+                "\nRescan the local blockchain for wallet related transactions.\n"
+                "\nArguments:\n"
+                "1. \"start_height\"    (numeric, optional) block height where the rescan should start\n"
+                "2. \"stop_height\"     (numeric, optional) the last block height that should be scanned\n"
```
Comment (Fuzzbawls) [inline src/wallet/rpcwallet.cpp:4190]: no quotes around numeric arguments

File: src/wallet/rpcwallet.cpp [hunk 3]
```diff
@@ -4179,6 +4179,67 @@ UniValue getsaplingnotescount(const JSONRPCRequest& request)
...
+                "\nResult:\n"
+                "{\n"
+                "  \"start_height\"     (numeric) The block height where the rescan has started. If omitted, rescan started from the genesis block.\n"
+                "  \"stop_height\"      (numeric) The height of the last rescanned block. If omitted, rescan stopped at the chain tip.\n"
```
Comment (Fuzzbawls) [inline src/wallet/rpcwallet.cpp:4194]:
```cpp
                "  \"start_height\": n,     (numeric) The block height where the rescan has started. If omitted, rescan started from the genesis block.\n"
                "  \"stop_height\": n      (numeric) The height of the last rescanned block. If omitted, rescan stopped at the chain tip.\n"
```

**Review by Fuzzbawls — CHANGES_REQUESTED** [review]
some initial styling changes left. also, prefer to have a blank line before the `Arguments:`, `Result:`, and `Examples:` sections for readability.

File: src/wallet/rpcwallet.cpp (hunk 1 above)
Comment (furszy) [reply src/wallet/rpcwallet.cpp:4186]: updated

**Comment by furszy** [comment]
updated per feedback, removed the quotes around numeric arguments.

**Review by random-zebra — APPROVED** [review]
ACK 896520ed6a2637b6a734d1faf3f9f2dc780252af

This needs a bit more work to be usable from the GUI console, due to the `cs_main` lock, which is held for the whole duration (essentially we need to get to bitcoin#11281... the rabbit hole starts with #2277).
Other than that, this works as intended from a headless node.

**Review by Fuzzbawls — APPROVED** [review]
ACK 896520ed6a2637b6a734d1faf3f9f2dc780252af

**Pull request merged**